Formula/duktape.rb

```diff
@@ -0,0 +1,38 @@
+class Duktape < Formula
+  desc "Embeddable Javascript engine with compact footprint"
+  homepage "http://duktape.org"
+  url "http://duktape.org/duktape-2.2.1.tar.xz"
+  sha256 "3abe2eed2553305262b892c98f550bb1a94cf4fd73b51dc5c176fe08e7ade7f2"
+
+  def install
+    inreplace "Makefile.sharedlibrary" do |s|
+      s.gsub! "-soname", "-install_name"
+      s.gsub! %r{\/usr\/local}, prefix
```

Comment:
Instead of editing the `Makefile`, just pass `INSTALL_PREFIX=#{prefix}` as one of the arguments to `make` down below.

Reply:
Unfortunately, there is no logic in the `Makefile.sharedlibrary` to accept the override from the command line.  The `INSTALL_PREFIX` is always set to `/usr/local`.

> Maybe I am missing something...

```
system "make", "INSTALL_PREFIX=#{prefix}", "-f", "Makefile.sharedlibrary", "install"
```

Reply:
There doesn't have to be logic in the Makefile for it: it's a feature of `make` that you can override any variable assignment from the command line like this.

Hmm. Works for me, using that exact `system "make", ...` call you have there. Don't know what's going on.

Reply:
apjanke, I appreciate the continued support.  I am going to work a bit on the upstream project to see if I can figure out the best approach.

If the fixes result in upstream PR's i'll likely close this PR until upstream is capable to support a clean homebrew-core implementation.

In the meantime, I worked through the **custom tap** approach to suit the project needs.

Reply:
No problem. Glad the custom tap worked out. Let me know if I can be of more assistance.

(FWIW, I suspect your problem with `INSTALL_PREFIX` may be a local configuration or usage issue on your part: their Makefile looks correct to me.)